### PR TITLE
Add Filter Command to DG (including activity diagram)

### PR DESCRIPTION
**Pull request by itsme-zeix**
Sequence diagram was omitted due to its complexity. An activity diagram is sufficient for understanding the implementation.

### CODE REVIEW

File: docs/images/del.png
Comment (iamdiluxedbutcooler) [inline docs/images/del.png:1]: How is this different from FilterActivityDiagramFilterAndDisplay.png? If it's being used, can consider changing the file name!

**Review by iamdiluxedbutcooler — APPROVED** [review]
Overall LGTM. Double check the logic and everything looks neatly written. Follow-up for the del.png file

File: docs/images/del.png
Comment (itsme-zeix) [reply docs/images/del.png:1]: Should be deleted! Will update now~

**Review by ZShunRen — APPROVED** [review]
LGTM

**Pull request merged**